src/core/tileLayer.js

```diff
@@ -920,9 +912,31 @@
 
           // mark the tile as covered
           this._setTileTree(tile);
-        }.bind(this));
+        } else {
+          tile.then(function () {
+            if (tile !== this.cache.get(tile.toString())) {
+              /* If the tile has fallen out of the cache, don't draw it -- it
```

Comment:
okay, makes sense. Do you have any statistics on how much it improved the performance?  but overall LGTM :+1: